### PR TITLE
Display traits for all types

### PR DESCRIPTION
**Pull request by jkilpatr**
This implements to upper and lower hex for Transaction and PrivateKey

### CODE REVIEW

File: src/private_key.rs [hunk 1]
```diff
@@ -173,8 +174,45 @@ impl<'de> Deserialize<'de> for PrivateKey {
             .and_then(move |bytes| PrivateKey::from_slice(&bytes))
             .map_err(serde::de::Error::custom)
 
-        // Ok(PrivateKey::from_slice(&bytes)?)
-        //
+        //Ok(PrivateKey::from_slice(&bytes)?)
```
Comment (mpapierski) [inline src/private_key.rs]: Commented out code can be safely removed

File: src/private_key.rs [hunk 2]
```diff
@@ -173,8 +174,45 @@ impl<'de> Deserialize<'de> for PrivateKey {
...
+    }
+}
+
+impl fmt::LowerHex for PrivateKey {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if f.alternate() {
+            let res = write!(f, "0x");
```
Comment (mpapierski) [inline src/private_key.rs]: TBH it could be shortened and simplified with just `write!(f, "0x")?`

File: src/signature.rs [hunk 3]
```diff
@@ -169,6 +167,26 @@ impl FromStr for Signature {
     }
 }
 
+impl fmt::LowerHex for Signature {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if f.alternate() {
+            write!(f, "0x{}", &self.to_string().to_lowercase())
```
Comment (mpapierski) [inline src/signature.rs]: ToString for Signature adds "0x" already so this could end up with `0x0x....`

File: src/private_key.rs (hunk 2 above)
Comment (jkilpatr) [reply src/private_key.rs]: nope, the 0x is only written when you pass the alternate modifier.

Comment (mpapierski) [reply src/private_key.rs]: Yeah I just meant the branches with `is_err()`. It could be simplified as just `write!(...)?`

**Pull request merged**